From 85c4fedfc9fe5a545dbb08395b0dec32053cb28c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sat, 15 Feb 2025 19:04:03 +0800
Subject: [PATCH 6/6] FROMLIST: qtwebengine: disable broken XNNPACK for ARM
 targets

The XNNPACK found in QtWebEngine uses bad NEON calls. Disable for now to
fix build.

Link: https://bugreports.qt.io/browse/QTBUG-129985
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 qtwebengine/src/core/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/qtwebengine/src/core/CMakeLists.txt b/qtwebengine/src/core/CMakeLists.txt
index 5ba2b00fc0..7c077674d0 100644
--- a/qtwebengine/src/core/CMakeLists.txt
+++ b/qtwebengine/src/core/CMakeLists.txt
@@ -544,6 +544,10 @@ foreach(arch ${archs})
                     )
                 endif()
             endif()
+            if(cpu MATCHES "arm.*")
+                # https://bugreports.qt.io/browse/QTBUG-129985
+                list(APPEND gnArgArg build_webnn_with_xnnpack=false)
+            endif()
             unset(cpu)
         endif()
 
-- 
2.48.1

